premailer/tests/test_cache.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import, unicode_literals
+
+import os
+import unittest
+import imp
+
+import cachetools
+
+
+class TestFunctionCache(unittest.TestCase):
+    def tearDown(self):
+        for key in (
+            "PREMAILER_CACHE",
+            "PREMAILER_CACHE_MAXSIZE",
+            "PREMAILER_CACHE_TTL",
+        ):
+            try:
+                del os.environ[key]
+            except KeyError:
+                pass
+
+    def test_unknown_cache(self):
+        os.environ["PREMAILER_CACHE"] = "UNKNOWN"
+
+        with self.assertRaises(Exception) as assert_context:
+            cache_fd = open("premailer/cache.py", "r")
+            imp.load_source("cache.py", "premailer/cache.py", cache_fd)
+
+        self.assertTrue(
+            assert_context.exception.args[0].startswith(
+                "Unsupported cache implementation"
+            )
+        )
+
+    def test_default_cache(self):
```

Comment:
I don't know if this test is really necessary. It's a deep implementation detail. The tests should really be more about the "user API" not about how things are done internally. The contents of this test method doesn't really ever mimic something a user would do.

Reply:
I agree with you. Removed.